staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -623,10 +653,63 @@ func preprocessDeletionListForMerging(key string, original map[string]interface{
 	return false, false, "", nil
 }
 
+// processReplaceKeys processes replaceKeys list.
+// When MergeParallelList is true, we validate the patch and
+// then clear the fields that are not in the list.
+func processReplaceKeys(original, patch map[string]interface{}, options MergeOptions) error {
+	replaceKeysInPatch, foundInPatch := patch[replaceKeysDirective]
+	if !foundInPatch {
+		return nil
+	}
+	delete(patch, replaceKeysDirective)
+
+	if !options.MergeParallelList {
+		replaceKeysInOriginal, foundInOriginal := original[replaceKeysDirective]
+		if foundInOriginal {
+			if !reflect.DeepEqual(replaceKeysInOriginal, replaceKeysInPatch) {
+				// This error actually should never happen.
+				return fmt.Errorf("%v and %v are not deep equal: this may happen when calculating the 3-way diff patch", replaceKeysInOriginal, replaceKeysInPatch)
+			}
+		} else {
+			original[replaceKeysDirective] = replaceKeysInPatch
+		}
+		return nil
+	}
+
+	replaceKeysList, ok := replaceKeysInPatch.([]interface{})
+	if !ok {
+		return mergepatch.ErrBadPatchFormatForReplaceKeys
+	}
+
+	// validate patch to make sure all field in patch is covered in replaceKeysList
```

Comment:
// // validate patch to make sure all fields in the patch are present in the retainKeysList